tmc-langs-qmake/src/test/java/fi/helsinki/cs/tmc/langs/qmake/QTestResultParserTest.java

```diff
@@ -0,0 +1,173 @@
+        QTestResultParser qtparser = createResultParser(failAndPass);
+        List<TestResult> results = qtparser.getTestResults();
+        assertEquals("There should only be two test results", 2, results.size());
+        TestResult result = results.get(0);
+        assertFalse("The test should be failing", result.isSuccessful());
+        assertEquals("The name of the test should be \"failing\"", "failing",
+                result.getName());
+        assertEquals("The message should contain the failed assertion",
+                "This test should've failed",
+                result.getMessage());
+        result = results.get(1);
+        assertTrue("The test should be passing", result.isSuccessful());
+        assertEquals("The name of the test should be \"passing\"", "passing",
+                result.getName());
+        assertEquals("The message should not contain any assertions",
+                "",
+                result.getMessage());
+    }
+
+    @Test
+    public void testTestRunWithOneFailingTestIsFailed() {
+        List<QTestCase> failAndPass = new ArrayList<>();
+        failAndPass.add(failing);
+        failAndPass.add(passing);
+        QTestResultParser qtparser = createResultParser(failAndPass);
+        RunResult result = qtparser.result();
+        assertEquals("Run result should be failed if one failing test",
+                result.status,
+                RunResult.Status.TESTS_FAILED);
+    }
+
+    @Test
+    public void testTestrunWithPassingTestsIsPassed() {
+        List<QTestCase> allPass = new ArrayList<>();
+        allPass.add(passing);
+        allPass.add(passing);
+        QTestResultParser qtparser = createResultParser(allPass);
+        RunResult result = qtparser.result();
+        assertEquals("Run result should be passed if all tests are passing",
+                result.status,
+                RunResult.Status.PASSED);
+    }
+
+    @Test
+    public void testParsedPointsMapToCorrectTests() {
+        List<QTestCase> allPass = new ArrayList<>();
+        allPass.add(passing);
+        allPass.add(another);
+
+        QTestResultParser qtparser = createResultParser(allPass);
+        List<TestResult> results = qtparser.result().testResults;
+        assertEquals(2, results.size());
+        assertEquals("Point for passing should be 1",
+                "1",
+                results.get(0).points.get(0));
+        assertEquals("Point for another should be 3",
+                "3",
+                results.get(1).points.get(0));
+    }
+
+    private QTestResultParser createResultParser(List<QTestCase> testCases) {
+        Path testPath = null;
+        try {
+            testPath = constructTestOutput(testCases);
+        } catch (Exception e) {
+            fail("Error creating or parsing output file: " + e.getMessage());
+        }
+        return new QTestResultParser(testPath);
+    }
+
+    private Path constructTestOutput(List<QTestCase> testCases) throws IOException {
```

Comment:
I don't fully understand what is happening here. Why are we testing against this generated output rather than real output?

Reply:
The test output XML files could be made static in the test resources, but every different edge case would have to have its own file. Inspiration to generate the test output came from the MakePlugin tests, where, I guess, those tests use the same reasoning.

Of course these generated test cases do not really test against the **true** output of the test framework, but the output is general enough (IMHO) to warrant generative output!

Reply:
I'm mostly concerned of a scenario where this code ends up generating output that is out-of-sync with the output of the real system. If that's not a concern, it's probably fine.

Reply:
Also, check out [`try`-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) to ensure we close the file handle in case of an exception.